### ISSUE TITLE
Defaulting country code if one isn't provided

### ISSUE THREAD

**Issue by tmiw** [Bouke/django-two-factor-auth#81]
Is it possible to have some sort of configuration variable to enable automatically defaulting to a particular country code if one isn't provided? For my site, the vast majority of the users are in the US and have US mobile numbers so it's confusing to require +1 at the beginning. Right now I added the following code to PhoneNumberForm in forms.py but obviously this isn't ideal:

```python
def clean_number(self):
    data = self.cleaned_data.get('number', '')
    if not data:
        raise forms.ValidationError("You must enter a phone number")
    if len(data) == 10 and (data[0:1] != "+" or data[0:2] != "00"):
        # Assume US number and prepend +1.
        data = "+1" + data
    return data
```

**Comment by Bouke** [comment]
Sorry for the late reply. I think the form could benefit from some UX improvements. The country code could be replaced with a predefined selection for example. The default value could either be a configurable setting, or auto-picked from some geo-mechanism.

**Comment by tmiw** [comment]
Geolocation to determine country code would be pretty nifty actually. It would even be supported in pretty much every modern browser too.

**Comment by Bouke** [comment]
Geo-location (client-side) would trigger warning dialogs, which might not be so nice. I was more thinking of something like Maxmind's GeoIP. Combined with a nice country-picker, I think it would make a better UX. What do you think?

**Comment by Bouke** [comment]
After some pondering, I think the difficulties are:

* The form is rendered by Python, so little influence on the HTML output possible.
* Geo-dependency should be avoided. It's a pain to install and the package should work as well without it.

Currently the validating is very simple; validating (a possible subset of) E.164 formatted phone numbers. The package [python-phonenumbers][1] (and its [lite version][2]) look like reasonable packages to improve the situation.

[1]: https://github.com/daviddrysdale/python-phonenumbers
[2]: https://pypi.python.org/pypi/phonenumberslite

**Comment by tmiw** [comment]
Client side geolocation could be done as opt-in (e.g. have to push a button to make it happen). One concern about python-phonenumbers is that there's 100MB of data that gets loaded the first time it's used, but the lite version might not have this issue.

**Comment by Bouke** [comment]
Resolved with PR #112.
